packages/compiler/server/serverlib.ts

```diff
@@ -531,6 +535,45 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function addImportCompletion(
+    program: Program,
+    document: TextDocument,
+    completions: CompletionList
+  ) {
+    const new_path = getPath(document);
+    const projectroot = await findProjectRoot(compilerHost, new_path);
+    const new_file = await loadFile(
+      compilerHost,
+      resolvePath(projectroot, "package.json"),
+      JSON.parse,
+      program.reportDiagnostic
+    );
+    let dep_list: string[] = [];
+    for (const val of new_file) {
```

Comment:
This loop is confusing. It took me a minute to understand what it is looping over. Then I found it: loadFile returns a 2-element tuple of the return value of JSON.parse and the SourceFile. 

I think you could do something llike:

``` TS
const [packageJson] = loadFile(...);
```

To take just the first element, then use packageJson.dependencies. We're never going to find dependencies on the SourceFile.

Reply:
I changed it to no longer traverse through the 2-element tuple